### PR TITLE
[TASK] Document options of the guides.xml

### PR DESCRIPTION
**Pull request by linawolf**
Also sort them by alphabet and give them more examples

### CODE REVIEW

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 1]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+    We advice to set the language to be used in each code-block however.
+
+..  _settings-guides-default-code-language-example:
+
+Example: Use `typoscript` as default langauge for code-blocks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+..  code-block:: xml
+    :caption: EXT:my_extension/Documentation/guides.xml (excerpt)
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:53]:
```suggestion
    :caption: EXT:my_extension/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4
```

Emphasize the line for better reception.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 2]
```diff
@@ -21,117 +17,227 @@ Example:
...
+    to the levels in the menu in some use cases.
+
+..  _settings-guides-max-menu-depth-example:
+
+Example: Limit the menu to 2 levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The changelog (documentation of the extension "core") contains several thousand
+pages. As very long lists of changelog entries in the menu are not helpful to
+the user and slow down the search
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:84]:
```suggestion
the user and slow down the search.
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 3]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+..  code-block:: xml
+    :caption: EXT:core/Documentation/guides.xml (excerpt)
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:87]:
```suggestion
    :caption: EXT:core/Documentation/guides.xml (excerpt)
    :emphasize-lines: 4
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 4]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+
 .. _settings-guides:
 
 The `<guides>` tag
 ==================
 
 It is used for general configuration during parsing and rendering.
 
+The following settings can be relevant for TYPO3 themed documentation:
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:31]: Please add that this is the root tag., so the structure is clear.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 5]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+Interlink mapping
+=================
+
+.. todo: describe interlink mapping more detailed
+
+A list of globally available Interlink (formerly "Intersphinx") repositories
+can be found in :ref:`Available default inventories <t3renderguides:available-default-inventories>`
+
+It is possible, though rarely needed to define custom interlink mappings:
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:108]: Maybe add information where to embed it, something like "is a direct child of the guides tag".

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 6]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+    <inventory id="sphinx" url="https://www.sphinx-doc.org/en/master/"/>
+
+
 .. _settings-guides-project:
 
 The `<project>` tag
 ===================
 
 This tag can contain the following meta information:
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:123]: Maybe add information where to embed it, something like "is a direct child of the guides tag".

**Review by brotkrueml — COMMENTED** [review]
First remarks, haven't finished review yet.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 7]
```diff
@@ -21,117 +17,227 @@ Example:
     a :file:`Documentation/Settings.cfg` you can use an automatic migration
     tool to :ref:`migrate the settings.cfg into a guides.xml <migrate_guides_xml>
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:18]: Here is missing closing backtick.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 8]
```diff
@@ -21,117 +17,227 @@ Example:
...
+    :type: string
+
+    Can be used to set the code-language that should be used for
+    :ref:`code-blocks <codeblock-directive>` if no language is set.
+
+    We advice to set the language to be used in each code-block however.
+
+..  _settings-guides-default-code-language-example:
+
+Example: Use `typoscript` as default langauge for code-blocks
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:49]: Example: Use `typoscript` as default language for code-blocks

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 9]
```diff
@@ -21,117 +17,227 @@ Example:
...
+--------------
+
+..  confval:: max-menu-depth
+    :name: guides-max-menu-depth
+    :type: integer
+    :Default: `PHP_INT_MAX`
+
+    Limits the main menu to a certain amount of levels. As level creation is
+    one of the main bottle-necks that can slow down rendering it can be helpful
+    to the levels in the menu in some use cases.
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:75]: to limit the levels in the menu in some use cases.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 10]
```diff
@@ -21,117 +17,227 @@ Example:
...
+
+..  _settings-guides-max-menu-depth-example:
+
+Example: Limit the menu to 2 levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The changelog (documentation of the extension "core") contains several thousand
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:82]: The changelog (documentation of the extension "core") consists of several thousands of

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 11]
```diff
@@ -21,117 +17,227 @@ Example:
...
+pages. As very long lists of changelog entries in the menu are not helpful to
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:83]: pages. The reason is that very long lists of changelog entries in the menu are not helpful to

File: Documentation/Images/ReferenceThisHeadlineDialog.png
Comment (Alagts) [inline Documentation/Images/ReferenceThisHeadlineDialog.png:1]: Why is this file blue?

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 12]
```diff
@@ -28,110 +28,218 @@ The `<guides>` tag
...
+    :type: string
+
+    Can be used to set the code-language that should be used for
+    :ref:`code-blocks <codeblock-directive>` if no language is set.
+
+    We advice to set the language to be used in each code-block however.
+
+..  _settings-guides-default-code-language-example:
+
+Example: Use `typoscript` as default langauge for code-blocks
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:49]:
```suggestion
Example: Use `typoscript` as default language for code-blocks
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 13]
```diff
@@ -28,110 +28,218 @@ The `<guides>` tag
...
+--------------
+
+..  confval:: max-menu-depth
+    :name: guides-max-menu-depth
+    :type: integer
+    :Default: `PHP_INT_MAX`
+
+    Limits the main menu to a certain amount of levels. As level creation is
+    one of the main bottle-necks that can slow down rendering it can be helpful
+    to the levels in the menu in some use cases.
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:76]:
```suggestion
    to limit the levels in the menu in some use cases.
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 14]
```diff
@@ -28,110 +28,218 @@ The `<guides>` tag
...
+
+..  _settings-guides-max-menu-depth-example:
+
+Example: Limit the menu to 2 levels
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The changelog (documentation of the extension "core") contains several thousand
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:83]:
```suggestion
The changelog (documentation of the extension "core") consists of several thousands of
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 15]
```diff
@@ -28,110 +28,218 @@ The `<guides>` tag
...
+pages. As very long lists of changelog entries in the menu are not helpful to
```
Comment (Alagts) [inline Documentation/GeneralConventions/GuidesXml.rst:84]:
```suggestion
pages. The reason is that very long lists of changelog entries in the menu are not helpful to
```

File: Documentation/GeneralConventions/GuidesXml.rst (hunk 4 above)
Comment (linawolf) [reply Documentation/GeneralConventions/GuidesXml.rst:32]: done

File: Documentation/GeneralConventions/GuidesXml.rst (hunk 5 above)
Comment (linawolf) [reply Documentation/GeneralConventions/GuidesXml.rst:115]: done

File: Documentation/GeneralConventions/GuidesXml.rst (hunk 6 above)
Comment (linawolf) [reply Documentation/GeneralConventions/GuidesXml.rst:133]: done

File: Documentation/Images/ReferenceThisHeadlineDialog.png
Comment (linawolf) [reply Documentation/Images/ReferenceThisHeadlineDialog.png:1]: woops, no idea, made a new screenshot, thanks for catching it!

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 16]
```diff
@@ -7,131 +7,252 @@ Configuration of the rendering - guides.xml
...
+
+.. _settings-guides-interlink-mapping:
+
+Interlink mapping
+=================
+
+.. todo: describe interlink mapping more detailed
+
+The :xml:`<inventory>` tag is located directly in the :ref:`guides tag <settings-guides>`.
+There can be 0 or more :xml:`<inventory>` tags, each defines one interlink
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:109]:
```suggestion
There can be 0 or more :xml:`<inventory>` tags, each defines one Interlink
```

Below, "Interlink" is written with a capitel "I", so this should be consistent.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 17]
```diff
@@ -7,131 +7,252 @@ Configuration of the rendering - guides.xml
...
-which the documentation refers.
+..  confval:: version
+    :name: guides-project-version
+    :type: string
 
-The version is shown below the title in the theme's release switch and in the
-title meta tag, the release is not shown currently  - but it should be
-kept anyway to satisfy internal requirements.
+    When documentation is rendered in the GitHub action for deployment onto
+    https://docs.typo3.org the version is set automatically to the extensions
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:190]:
```suggestion
    https://docs.typo3.org the version is set automatically to the extension's
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 18]
```diff
@@ -7,131 +7,252 @@ Configuration of the rendering - guides.xml
...
+    version derived from the git tag. Setting the version in the
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:191]:
```suggestion
    version derived from the Git tag. Setting the version in the
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 19]
```diff
@@ -7,131 +7,252 @@ Configuration of the rendering - guides.xml
...
 
-If you do not need the separation provided between version and release,
-just set them both to the same value.
+    version
+        The major project version, used as the replacement for :rst:`|version|`.
+        For example this may be something like 12.4.
+
+    release
+        The full project version, used as the replacement for :rst:`|release|`
+        For example 13.0.0-dev.
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:208]:
```suggestion
        For example 12.4.15-dev.
```

version and release should correspond, the current example does not make sense.

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 20]
```diff
@@ -155,18 +276,151 @@ For example:
...
+    The interlink shortcode will be displayed in the "Reference this headline"
+    dialog when users click on the link symbol beside a headline or other
+    linkable element.
+
+    These references can be used to link to your manual from other documents
+    rendered with the TYPO3 theme. However they will only work if you follow
+    these rules:
 
-.. _settings-guides-footer-links:
+    *   TYPO3 third party extensions: composer name, example: `georgringer/news`
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:395]:
```suggestion
    *   TYPO3 third-party extensions: Composer name, example: `georgringer/news`
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 21]
```diff
@@ -155,18 +276,151 @@ For example:
...
+    *   System extensions: composer name, example: `typo3/cms-adminpanel`
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:396]:
```suggestion
    *   System extensions: Composer name, example: `typo3/cms-adminpanel`
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 22]
```diff
@@ -175,59 +429,191 @@ project_home
...
-..  code-block:: xml
-    :caption: Documentation/guides.xml, excerpt
+.. _settings-guides-project-links-example:
 
-    <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
-       project-home="https://docs.typo3.org/m/typo3/docs-how-to-document/main/en-us/"
-       project-contact="https://typo3.slack.com/archives/C028JEPJL"
-       project-repository="https://github.com/TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument"
-       project-issues="https://github.com/TYPO3-Documentation/TYPO3CMS-Guide-HowToDocument/issues"
+Example: Project links for a third party extension
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:494]:
```suggestion
Example: Project links for a third-party extension
```

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 23]
```diff
@@ -175,59 +429,191 @@ project_home
...
+    :Default: `stable`
+
+    You can set the preferred TYPO3 Core version. This version is used to determine
+    the desired version for links to other manuals and system extensions.
+
+    ..  code-block:: xml
+        :caption: Documentation/guides.xml
+
+        <extension class="\T3Docs\Typo3DocsTheme\DependencyInjection\Typo3DocsThemeExtension"
+                   typo3-core-preferred="12.4" />
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:571]:
```suggestion
                   typo3-core-preferred="stable" />
```

I suggest to use "stable" as people tend to copy/paste those and will come up later with references to outdated manuals (when not adapted manually).

File: Documentation/GeneralConventions/GuidesXml.rst [hunk 24]
```diff
@@ -175,59 +429,191 @@ project_home
...
+    </guides>
+
+Example: Always link to the latest stable TYPO3 version
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you want your links to always go to the latest stable long time support version
+(LTS) of the TYPO3 Core, you can set :confval:`guides-extension-typo3-core-preferred` to stable.
+
+At the time of writing references to official manuals and Core extensions will
+be rendered to version 12.4. Once TYPO3 13LTS is released they will automatically
```
Comment (brotkrueml) [inline Documentation/GeneralConventions/GuidesXml.rst:605]:
```suggestion
be rendered to version 12.4. Once TYPO3 13 LTS is released they will automatically
```

**Review by brotkrueml — APPROVED** [review]

**Pull request merged**